fame/core/module_dispatcher.py

```diff
@@ -82,6 +98,26 @@ def next_module(self, types_available, module_name, excluded_modules):
             else:
                 return self._shortest_path_to_module(types_available, module, excluded_modules)
 
+    def get_next_preloading_module(self, types_available, excluded_modules):
+        candidate_modules = []
+        for type_ in types_available:
+            candidate_modules += self.get_preloading_modules_for(type_)
+
+        result = []
+
+        for module in candidate_modules:
+            if module.info['name'] in excluded_modules:
+                continue
+
+            for acts_on in module.info['acts_on']:
+                if acts_on in types_available:
+                    result.append(module.info['name'])
+
+        if len(result) > 0:
+            return result[0]
+        else:
+            return ""
```

Comment:
Seems quite complicated since `hash` is the only possible type. Or am I missing something ?

Reply:
No, you are right. See my other comment on simplifiying PreloadingModules.